packages/angular_devkit/build_angular/test/browser/svg_spec_large.ts

```diff
@@ -0,0 +1,59 @@
+/**
+ * @license
+ * Copyright Google Inc. All Rights Reserved.
+ *
+ * Use of this source code is governed by an MIT-style license that can be
+ * found in the LICENSE file at https://angular.io/license
+ */
+
+import { runTargetSpec } from '@angular-devkit/architect/testing';
+import { join, normalize, virtualFs } from '@angular-devkit/core';
+import { tap } from 'rxjs/operators';
+import { browserTargetSpec, host, outputPath } from '../utils';
+
+describe('Browser Builder allow svg', () => {
+
+  beforeEach(done => host.initialize().toPromise().then(done, done.fail));
+  afterEach(done => host.restore().toPromise().then(done, done.fail));
+
+  it('works with aot',
+    (done) => {
+
+    const svg = `
+      <svg xmlns="http://www.w3.org/2000/svg">
+        <text x="20" y="20" font-size="20" fill="red">Hello World</text>
+      </svg>`;
+
+    host.writeMultipleFiles({
+      './src/app/app.component.svg': svg,
+      './src/app/app.component.ts': `
+        import { Component } from '@angular/core';
+
+        @Component({
+          selector: 'app-root',
+          templateUrl: './app.component.svg',
+          styleUrls: []
+        })
+        export class AppComponent {
+          title = 'app';
+        }
+      `,
+    });
+
+    const overrides = { aot: true };
+
+    runTargetSpec(host, browserTargetSpec, overrides).pipe(
+      tap((buildEvent) => expect(buildEvent.success).toBe(true)),
+      tap(() => {
+        const content = virtualFs.fileBufferToString(
+          host.scopedSync().read(join(outputPath, 'main.js')),
+        );
+
+        expect(content).toContain('":svg:svg"');
```

Comment:
Where does `":svg:svg"` come from btw? I don't see it in the svg content.

Reply:
That's part of the code that is generated by the Angular for the .svg template.

For example, when I have the following .svg file
```
<svg xmlns="http://www.w3.org/2000/svg">
  <text x="20" y="20" font-size="20" fill="red">Hello World</text>
</svg>
```
then the Angular template compiler generates this code:
```
function View_SvgComponentComponent_0(_l) { return _angular_core__WEBPACK_IMPORTED_MODULE_1__["ɵvid"](0, [(_l()(), _angular_core__WEBPACK_IMPORTED_MODULE_1__["ɵeld"](0, 0, null, null, 2, ":svg:svg", [["xmlns", "http://www.w3.org/2000/svg"]], null, null, null, null, null)), (_l()(), _angular_core__WEBPACK_IMPORTED_MODULE_1__["ɵeld"](1, 0, null, null, 1, ":svg:text", [["fill", "red"], ["font-size", "20"], ["x", "20"], ["y", "20"]], null, null, null, null, null)), (_l()(), _angular_core__WEBPACK_IMPORTED_MODULE_1__["ɵted"](-1, null, ["Hello World"]))], null, null); }
```
I don't know exactly how the template compiler and view engine work. For my test, I just looked at the generated code when I manually tested svg support, and thought that checking for "svg:svg" would be a good way to test if Angular actually used the .svg file as template.

Without the changes from this PR, the same .svg used as template would generate the following code:
```
function View_SvgComponentComponent_0(_l) { return _angular_core__WEBPACK_IMPORTED_MODULE_1__["ɵvid"](0, [(_l()(), _angular_core__WEBPACK_IMPORTED_MODULE_1__["ɵted"](-1, null, ["svg-component.component.svg"]))], null, null); }
```
It just outputs the file name as string.

I hope that answers your question :)

Reply:
It does answer my question, yes. I think we'll need to alter this test for when we move to the Ivy compiler. But as of now it doesn't need to change.